Setup for project

SinaElahimanesh opened the pull request:
Add a piece of headers and first setup of project

SinaElahimanesh committed:
create the design of the project and packages and files
+ add some headers to files
+ search about sending a directory with socket - comment a piece of code about it to make it better at FileSender.cpp

saeed1gorji commented:
لطفا دقت کنید که تمامی کدهای شما باید روی شاخه‌ی گیت خودتان باشد
در حال حاضر کدهای شما و همچنین پول ریکوئست فعلی ثبت شده همگی روی شاخه‌ی مستر هستند که قابلیت مرج کردن برای آن وجود ندارد.

لطفا علاوه بر انتقال به شاخه‌ی مخصوص به خودتان موارد زیر را نیز اصلاح نمایید:

- پوشه بندی پروژه خوب است، در تکمیل آن دو پوشه‌ی اصلی زیر را اضافه کنید:
src
test
- ساختار فعلی را در پوشه‌ی زیر قرار دهید:
src
- در پوشه‌ی تست فعلا نیازی نیست فایلی اضافه شود و همین که به صورت خالی وجود داشته باشد کفایت می‌کند.
- واسط را تکمیل کنید. یعنی تمام متودهای لازم تا جایی که قابل پیش بینی است در کلاس ها اعلان شوند و همچنین پارامترهای ورودی آن‌ها نیز طبق پیش بینی فعلی تکمیل شود.
- برای استفاده از فایل سیستم لطفا از کتابخانه‌ی بوست فایل سیستم استفاده شود.
- سایر کتابخانه‌هایی که در آینده برای استفاده اضافه می‌شوند نسخه‌ی سی پلاس پلاس آن در صورت وجود به نسخه‌ی سی ارجحیت دارد.
- فایل‌های پیاده‌سازی با پسوند cpp
در این مرحله لازم نیستند. این فایل‌ها را حذف کنید تا در مراحل بعدی اضافه شوند. این کار به جهت حفظ نظم در روند توسعه پروژه می‌باشد.

saeed1gorji commented:
لطفا پس از اتمام کار اعلام کنید تا بررسی شود.

SinaElahimanesh commented:
> لطفا پس از اتمام کار اعلام کنید تا بررسی شود.

با سلام.
 بله حتما

SinaElahimanesh committed:
create other necessary fields and methods at header files

SinaElahimanesh commented:
> لطفا پس از اتمام کار اعلام کنید تا بررسی شود.

شبتون به خیر
موارد ذکر شده رو اعمال کردم و با جستجو درباره ی فایل سیستم بوست فیلدها ومتدهای لازم را اضافه کردم
و ساختار پروژه را هم اصلاح کردم. 
لطفا اصلاحات انجام شده رو توی شاخه ی گیت خودم ملاحظه بفرمایید اگر مشکلات آن را ذکر کنید ممنون میشم.

saeed1gorji commented:
ممنون.
لطفا ضمن اصلاح مواردی که در ادامه ذکر شده است یک پول ریکوئست جدید برای شاخه‌ی مرتبط ایجاد کنید تا ادامه‌ی بحث را آنجا پیگیری کنیم و همچنین تغییرات شاخه شما را با امکانات گیت هاب مشاهده کنم.

- لطفا سرآیند زیر را از کامپرشن حذف کنید:
dirent.h
- کامنت ها را با سبک داکسیجن بازنویسی کنید. به انگلیسی:
doxygen
- کامنت هر عضو کلاس در یک خط جداگانه و بالای هر کدام از عضو ها قرار می‌گیرد و همچنین بین اعضا یک خط خالی برای جداسازی آن‌ها از یکدیگر قرار بگیرد
- عبارت پکیج از نام دایرکتوری ها حذف شود. سبک نام گذاری دایرکتوری‌های پروژه به این صورت است که کل نام با حروف کوچک نوشته شده و کلمات فقط با خط تیره از هم جدا شوند. نام خود فایل‌ها خوب است و به همین صورت درست است

SinaElahimanesh commented:
> ممنون.
> لطفا ضمن اصلاح مواردی که در ادامه ذکر شده است یک پول ریکوئست جدید برای شاخه‌ی مرتبط ایجاد کنید تا ادامه‌ی بحث را آنجا پیگیری کنیم و همچنین تغییرات شاخه شما را با امکانات گیت هاب مشاهده کنم.
> 
> * لطفا سرآیند زیر را از کامپرشن حذف کنید:
>   dirent.h
> * کامنت ها را با سبک داکسیجن بازنویسی کنید. به انگلیسی:
>   doxygen
> * کامنت هر عضو کلاس در یک خط جداگانه و بالای هر کدام از عضو ها قرار می‌گیرد و همچنین بین اعضا یک خط خالی برای جداسازی آن‌ها از یکدیگر قرار بگیرد
> * عبارت پکیج از نام دایرکتوری ها حذف شود. سبک نام گذاری دایرکتوری‌های پروژه به این صورت است که کل نام با حروف کوچک نوشته شده و کلمات فقط با خط تیره از هم جدا شوند. نام خود فایل‌ها خوب است و به همین صورت درست است

با سلام
در شاخه ی مرتبط پل ریکویست را انجام داده بودم. 
بله حتما اصلاحات لاطمه در ساختار پروژه را اعمال میکنم و در ضمن با مطالعخ در مرود داکسیجن شیوه ی کامنت گذاری را اصلاح میکنم.
باتشکر

saeed1gorji commented:
بسیار عالی. من پول ریکوئست جدید شما را پیدا نکردم. در پروفایل شما یک پول ریکوئست نشان داده می‌شود. اگر ساخته شده لطفا لینک را اینجا ارسال کنید

SinaElahimanesh committed:
edit the structure of project by changing the name of packages to standard naming convention
+ edit some extra includes and remove them all from header files
+ change the commenting of project to the doxygen standard and make the whole project well commented!

SinaElahimanesh commented:
> بسیار عالی. من پول ریکوئست جدید شما را پیدا نکردم. در پروفایل شما یک پول ریکوئست نشان داده می‌شود. اگر ساخته شده لطفا لینک را اینجا ارسال کنید

همه ی مشکلات وارده را اصلاح کردم و سعی کردم تا حد امکان با تمیز و کامل نوشتن تمام موارد و همچنین کامنت گذاری مناسب به سبک داکسیجن از مشکلات احتمالی موجود جلوگیری کنم.
باتشکر

https://github.com/SinaElahimanesh/cplusclass1399/pull/2

saeed1gorji commented:
پول ریکوئست ذکر شده برای مرج کردن شاخه‌های مخزن خودتان است (که نیاز به تایید کسی ندارد و دسترسی کامل آن در اختیار خودتان است). لطفا پول ریکوئست را برای مرج با شاخه‌ی موجود در مخزن شرکت ایجاد کنید.

یعنی به این صورت:
`SinaElahimanesh:Sina_Elahimanesh --> mahsanco:Sina_Elahimanesh`

SinaElahimanesh commented:
> پول ریکوئست ذکر شده برای مرج کردن شاخه‌های مخزن خودتان است (که نیاز به تایید کسی ندارد و دسترسی کامل آن در اختیار خودتان است). لطفا پول ریکوئست را برای مرج با شاخه‌ی موجود در مخزن شرکت ایجاد کنید.
> 
> یعنی به این صورت:
> `SinaElahimanesh:Sina_Elahimanesh --> mahsanco:Sina_Elahimanesh`

با سلام
ببخشید ولی این دسترسی نمیده شاخه ی دیگه ای هم نیست جزو چیزایی که میتونم ببینم
Push failed: remote: Permission to mahsanco/cplusclass1399.git denied to SinaElahimanesh. unable to access 'https://github.com/mahsanco/cplusclass1399.git/'

saeed1gorji commented:
لطفا این بحث را در پرسش و پاسخ یا چت مایکروسافت تیمز مطرح کنید تا مشکل شما بررسی شود زیرا ممکن است طولانی باشد.

در صورتی که ویدیوی قرار داده شده در تیمز را مشاهده نکرده اید حتما آن را مرور کنید. همچنین درباره ساخت شاخه‌ی جدید در گیت هاب جست و جو نمایید و یک ویدیوی کوتاه آموزشی برای آن ببینید. اگر مشکل حل نشد در تیمز مطرح کنید.

rezaebrh commented:
به دلیل اتمام موعد تحویل این مسئله و عدم پیگیری انجام دهنده اصلی، این مسئله بسته شد.
